packages/@aws-cdk/aws-apigatewayv2/README.md

```diff
@@ -19,6 +19,100 @@
+
+For more information about WebSocket APIs, see [About WebSocket APIs in API Gateway](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-overview.html) in the _API Gateway Developer Guide_. For more information about HTTP APIs, see [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api.html) in the _API Gateway Developer Guide_.
+
+To create [REST APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-rest-api.html), use `@aws-cdk/aws-apigateway` instead.
+
+
+### HTTP API
+
+`CfnApi` is the L1 construct to create either `WebSocket APIs` or `HTTP APIs`. At this moment, `HTTP APIs` supports both `Lambda Proxy Integration` and `HTTP Proxy Integration`. See [Working with AWS Lambda Proxy Integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) and [Working with HTTP Proxy Integrations for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-http.html) for more information.
+
+
+To create `HTTP APIs` with `Lambda Proxy Integration`, simply use  `LambdaProxyApi`.
+
+
+```ts
+// Create a HTTP API with Lambda Proxy Integration as its $default route
+const api = new apigatewayv2.LambdaProxyApi(stack, 'LambdaProxyApi', {
+  handler
+});
+```
+
+To create `HTTP APIs` with `HTTP Proxy Integration`, use `HttpProxyApi` instead.
+
+```ts
+// Create a HTTP API with HTTP Proxy Integration
+new apigatewayv2.HttpProxyApi(stack, 'HttpProxyApi', {
+  url: 'https://aws.amazon.com'
+});
+```
+
+
+### WebSocket API
+
+The WebSocket API is not implemented yet in this L2 construct library, however, as `Api` class is provided, it's still possible to create the Websocket APIs with the `Api` class.
+
+
+## Route
+
+`Routes` direct incoming API requests to backend resources. `Routes` consist of two parts: an HTTP method and a resource path—for example, `GET /pets`. You can define specific HTTP methods for your route, or use the `ANY` method to match all methods that you haven't defined for a resource. You can create a `$default route` that acts as a catch-all for requests that don’t match any other routes. See [Working with Routes for HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-routes.html).
+
+When you create HTTP APIs with either `Lambda Proxy Integration` or `HTTP Proxy Integration`, the `$default route` will be created as well.
+
+To create a specific route for an existing `Api` resource, specify the `httpPath` and an optional `httpMethod` property.
+
 ```ts
-import apigatewayv2 = require('@aws-cdk/aws-apigatewayv2');
+// create a specific 'GET /some/very/deep/route/path' route with Lambda proxy integration for an existing HTTP API
+const someDeepLambdaRoute = new apigatewayv2.LambdaRoute(stack, 'SomeLambdaRoute', {
+  api,
+  handler,
```

Comment:
Add two lines one declaring the HTTP API and the other declaring the lambda function (as shown in a previous comment)